plugins/file/plugin.go

```diff
@@ -125,24 +192,124 @@ func setupFile6(args ...string) (handler.Handler6, error) {
-	if len(args) < 1 {
-		return nil, nil, errors.New("plugins/file: need a file name")
+	if v6 {
+		if len(args) < 1 {
+			return nil, nil, errors.New("plugins/file: need a file name")
+		}
+		filename := args[0]
+		if filename == "" {
+			return nil, nil, errors.New("plugins/file: got empty file name")
+		}
+		records, err := LoadDHCPv6Records(filename)
+		if err != nil {
+			return nil, nil, fmt.Errorf("plugins/file: failed to load DHCPv6 records: %v", err)
+		}
+		log.Printf("plugins/file: loaded %d leases from %s", len(records), filename)
+		StaticRecords = records
+	} else {
+		if len(args) < 4 {
+			return nil, nil, errors.New("plugins/file: need a file name, server IP, netmask and a lease time")
+		}
+		filename = args[0]
+		if filename == "" {
+			return nil, nil, errors.New("plugins/file: got empty file name")
+		}
+		ipRangeStart = net.ParseIP(args[1])
+		if ipRangeStart.To4() == nil {
+			return nil, nil, errors.New("plugins/file: expected an IP address, got: " + args[1])
+		}
+		ipRangeEnd = net.ParseIP(args[2])
+		if ipRangeEnd.To4() == nil {
+			return nil, nil, errors.New("plugins/file: expected an IP address, got: " + args[2])
+		}
+		if binary.BigEndian.Uint32(ipRangeStart.To4()) >= binary.BigEndian.Uint32(ipRangeEnd.To4()) {
+			return nil, nil, errors.New("plugins/file: start of IP range has to be lower than the end fo an IP range")
+		}
+		var err error
+		LeaseTime, err = time.ParseDuration(args[3])
+		if err != nil {
+			return Handler6, Handler4, errors.New("plugins/file: expected an uint32, got: " + args[3])
+		}
+		records, err := LoadDHCPv4Records(filename)
+		if err != nil {
+			return nil, nil, fmt.Errorf("plugins/file: failed to load DHCPv4 records: %v", err)
+		}
+		DHCPv4Records = records
+
+		rand.Seed(time.Now().Unix())
+
+		log.Printf("plugins/file: loaded %d leases from %s", len(DHCPv4Records), filename)
+	}
+
+	return Handler6, Handler4, nil
+}
+func createIP(rangeStart net.IP, rangeEnd net.IP) (*Record, error) {
+	ip := make([]byte, 4)
+	rangeStartInt := binary.BigEndian.Uint32(rangeStart.To4())
+	rangeEndInt := binary.BigEndian.Uint32(rangeEnd.To4())
+	binary.BigEndian.PutUint32(ip, random(rangeStartInt, rangeEndInt))
+	taken := checkIfTaken(ip)
+	for taken {
+		ipInt := binary.BigEndian.Uint32(ip)
+		ipInt++
+		binary.BigEndian.PutUint32(ip, ipInt)
+		if ipInt > rangeEndInt {
+			break
+		}
+		taken = checkIfTaken(ip)
+	}
+	for taken {
+		ipInt := binary.BigEndian.Uint32(ip)
+		ipInt--
+		binary.BigEndian.PutUint32(ip, ipInt)
+		if ipInt < rangeStartInt {
+			return &Record{}, errors.New("plugins/file: no new IP addresses available")
+		}
+		taken = checkIfTaken(ip)
 	}
-	filename := args[0]
-	if filename == "" {
-		return nil, nil, errors.New("plugins/file: got empty file name")
+	return &Record{IP: ip, expires: time.Now().Unix() + int64(LeaseTime.Seconds())}, nil
+
+}
+func random(min uint32, max uint32) uint32 {
+	return rand.Uint32()%(max-min+1) + min
```

Comment:
this function suffers of modulo bias, you should use `rand.Intn` instead, https://golang.org/pkg/math/rand/#Intn